packages/flutter_tools/test/general.shard/devtools_launcher_test.dart

```diff
@@ -34,13 +35,56 @@ void main() {
     );
   });
 
+   testWithoutContext('DevtoolsLauncher does not launch devtools if unable to reach pub.dev', () async {
+    final DevtoolsLauncher launcher = DevtoolsServerLauncher(
+      pubExecutable: 'pub',
+      logger: logger,
+      platform: platform,
+      persistentToolState: persistentToolState,
+      httpClient: FakeHttpClient.list(<FakeRequest>[
+        FakeRequest(
+          Uri.https('pub.dev', ''),
+          method: HttpMethod.head,
+          response: const FakeResponse(statusCode: HttpStatus.internalServerError),
+        ),
+      ]),
+      processManager: FakeProcessManager.list(<FakeCommand>[]),
+    );
+
+    final DevToolsServerAddress address = await launcher.serve();
+    expect(address, isNull);
+  });
+
+  testWithoutContext('DevtoolsLauncher pings PUB_HOSTED_URL instead of pub.dev for online check', () async {
```

Comment:
Test if the `PUB_HOSTED_URL` is malformed?

Reply:
Done